common-libs/common/src/Affinity.ts

```diff
@@ -624,21 +625,21 @@ export class Affinity {
 
   private _buildFragment<TKeys extends string, TData extends SimpleThing & Record<TKeys, unknown>>(
     credential: VCV1<VCV1Subject<TData>>,
-    fields: TKeys[],
+    fields: string[],
   ) {
     if (Array.isArray(credential.credentialSubject)) {
       throw new Error()
     }
 
-    const dataFields: Record<TKeys, Record<string, never>> = {} as any
-    for (const field of fields) {
-      if (credential.credentialSubject.data[field] === undefined) {
-        throw new Error(`Field "${field}" not a part of credential`)
-      }
-
-      dataFields[field] = {}
+    const validationErrors = validateObjectHasPaths(credential.credentialSubject.data, fields)
+    if (validationErrors) {
+      const firstError = validationErrors[0]
+      throw new Error(`Field "${firstError.field}" not a part of credential`)
     }
 
+    const dataFields = buildObjectSkeletonFromPaths(fields)
```

Comment:
What's the advantage of implementing on our own over using an existing library like https://www.npmjs.com/package/json-mask ?

Reply:
I updated syntax to json-mask variant ('/' symbol instead of '.')
I haven't found any useful solution as all of them have specific usage and in case of json-mask it generates specific object, not equal what we need, and it will require **more code** to convert json-mask.compile output with jsonld-signatures-bbs.deriveProof input